astropy/coordinates/earth.py

```diff
@@ -667,14 +679,24 @@ def get_gcrs(self, obstime):
         """
         # do this here to prevent a series of complicated circular imports
         from .builtin_frames import GCRS
+        return self._in_alternate_frame(GCRS, obstime)
 
-        itrs = self.get_itrs(obstime)
-        # Assume the observatory itself is fixed on the ground.
-        # We do a direct assignment rather than an update to avoid validation
-        # and creation of a new object.
-        zeros = np.broadcast_to(0. * u.km / u.s, (3,) + itrs.shape, subok=True)
-        itrs.data.differentials['s'] = CartesianDifferential(zeros)
-        return itrs.transform_to(GCRS(obstime=obstime))
+    def get_cirs(self, obstime):
```

Comment:
This works, and I like the refactoring with `_in_alternate_frame`, but I wonder if we shouldn't try to avoid a proliferation of methos...

One option to do so would be to make `EarthLocation` a bit more like a frame and define a `.transform_to(frame)` method which is the same as your `_in_alternate_frame, except that it takes a frame instance rather than a frame class (i.e., the `obstime` should already be done by the user).

Another option would be to give `get_itrs` a new keyword argument that asks to include a zero velocity component.  Then, inside code a user can just to `el.get_itrs(obstime, include_velocity=True).transform_to(...)`

Reply:
You will also need a `get_tete_posvel()` method if TETE is converted to use an `EarthLocation`. The difference being the use of greenwich apparent sidereal time rather than earth rotation angle.

Reply:
Hmm, with TETE also in play, all the more reason to try to find a more general way to do this...

Reply:
My suggestion for implementing these functions in `EarthLocation` was just a matter of convenience for use in the relevant intermediate rotation transforms that require conversion of an `EarthLocation` to GCRS position and velocity. Of course, you can just use the ITRS position from the `EarthLocation` and convert it with the `itrs_to_cirs/tete` matrix, but then you also need to calculate the relevant velocity.

Reply:
From the new `_in_alternate_frame()` private method, it seems clear that all that is really needed is to provide a way to have an ITRS with associated velocities of 0 -- after that, the usual transformation machinery suffices. So, in that sense, I don't think we need a whole lot of extra methods.